k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/selectedstate/command/UidFetchCommandTest.java

```diff
@@ -106,8 +107,8 @@ private UidFetchCommand createUidFetchCommand(int maximumAutoDownloadMessageSize
         return new UidFetchCommand.Builder()
                 .maximumAutoDownloadMessageSize(maximumAutoDownloadMessageSize)
                 .idSet(Collections.singletonList(uid))
-                .messageParams(fetchProfile, Collections.singletonMap(String.valueOf(uid),
-                        (Message) createImapMessage(String.valueOf(uid))))
+                .messageParams(fetchProfile, new HashMap<>(Collections.singletonMap(String.valueOf(uid),
```

Comment:
a builder shouldn't take mutable data structures. if this parameter actually needs a mutable hashmap, rather than an immutable Map type, something is wrong